### ISSUE TITLE
Captions with hyperlinks have a dark colour that's not compatible with the black captions banner - Twenty Twenty One theme

### ISSUE THREAD

**Issue by jamesozzie** [Automattic/amp-wp#6734]
### Bug Description

When there is a caption with a hyperlink in a gallery block this hyperlink is unreadable when it's displayed as a carousel on AMP URLs. This is only applicable for the Twenty Twenty One theme and [was identified by one user in the support forums](https://wordpress.org/support/topic/amp-carousel-link-color-not-readable-on-desktop/#post-15092423).

[Recording of this experience.](https://recordit.co/2Iop7ICt97) 

Gif of experience below:

![2021-11-22-16-12-19](https://user-images.githubusercontent.com/41326532/142896129-8a603678-d919-4bc1-afaf-2a1fb9fa2c83.gif)

This is not an issue with the Twenty Twenty or Twenty Nineteen themes, with the [hyperlink font color white and underlined](https://i.imgur.com/CDcYhJk.png
). The same approach would be suitable for Twenty Twenty One.

  

### Expected Behaviour

The figcaption color should be more readable. 

The below CSS rule works as a temporary solution:

```
figcaption.amp-wp-gallery-caption a {
    color: #fff;
}
```

### Plugin Version

2.2.0 Alpha

### AMP plugin template mode

Standard

### WordPress Version

5.8

**Comment by westonruter** [comment]
The `.amp-wp-gallery-caption` currently has this white color alread:

https://github.com/ampproject/amp-wp/blob/2e31d810c4945474bd16a4278f7bb781401b0ffb/assets/css/src/amp-default.css#L54-L63

So yeah, I think it's just a matter of adding:

```css
amp-carousel .amp-wp-gallery-caption a {
    color: #fff; 
}
```

**Comment by bartoszgadomski** [comment]
I think this is more a Gutenberg stylesheet issue rather than AMP, as the same appears in Block editor:

|Twenty Twenty-Two theme (editor)|
|--|
|![Screenshot 2021-12-1 at 13 09 28](https://user-images.githubusercontent.com/4476915/144232262-700ae84b-c6de-4351-a80b-df3c1e6acf3b.png)|

For Twenty Twenty theme, this issue is resolved in editor only, the same black link text appears on front-end:

|Twenty Twentytheme (editor)|Twenty Twentytheme (front)|
|--|--|
|![Screenshot 2021-12-1 at 13 22 24](https://user-images.githubusercontent.com/4476915/144234141-c885fb3c-79f0-460d-9cc0-9f139c28bf33.png)![Screenshot 2021-12-1 at 13 22 31](https://user-images.githubusercontent.com/4476915/144234157-47b4f740-5f7a-43df-8499-0eb50d163371.png)|![Screenshot 2021-12-1 at 13 23 06](https://user-images.githubusercontent.com/4476915/144234686-b48f4eff-70b2-406b-a4ba-a5821333c69f.png)|

I think could suggest adding new CSS property here: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/gallery/style.scss#L48-L66, something like:

```
figcaption {
    [...]

    a {
        color: inherit;
    }
}
```

This should resolve reported bug as well.

**Comment by bartoszgadomski** [comment]
Gutenberg issue: https://github.com/WordPress/gutenberg/issues/37032
PR with fix: https://github.com/WordPress/gutenberg/pull/37033

**Comment by westonruter** [comment]
What about when the gallery has “Display as carousel” enabled?

![image](https://user-images.githubusercontent.com/134745/144343946-3d9ccae9-68ca-47f9-b1de-74c8cd505fe4.png)

I think that's the issue here. However, it seems that the new carousel is not working with the latest Gallery block in WordPress 5.9-beta (and the latest Gutenberg).

Could you double check that? Try enabling carousel on a Gallery block in WordPress 5.8, and then try doing the same in WordPress 5.9-beta1 (or the latest Gutenberg). Set the gallery to link to the attachment page so that the captions get hyperlinks.

**Comment by maitreyie-chavan** [comment]
@delawski please could you take a look at this and see if any further updates need to be made by us?

**Comment by delawski** [comment]
@maitreyie-chavan There still was an issue with the caption colors. It's addressed and now waits for a code review (#6907).

**Comment by pooja-muchandikar** [comment]
QA Passed ✅ 

Captions with hyperlinks looks better.
Themes Tested: Twenty Twenty One, Twenty Twenty Two and Twenty Twenty

**Twenty Twenty One**
![2021](https://user-images.githubusercontent.com/41000648/159434737-9f01db8a-1893-4a83-b42c-170fe08972a2.png)

---

**Twenty Twenty Two**
![2022](https://user-images.githubusercontent.com/41000648/159434784-da7a61c1-ddd7-419d-9d1f-94f60c4bf3aa.png)

---

**Twenty Twenty**
![2020](https://user-images.githubusercontent.com/41000648/159434828-cd00b645-de3c-4319-8441-a6cfe3ebe829.png)
